### DISCUSSION TITLE
IUS 2023 vbeam discussion

### DISCUSSION THREAD

**Discussion by magnusdk** [magnusdk/vbeam#8] [Q&A]
Hi! 👋 I'm happy to answer any questions here.

If you want to chat in-person, I am also at the poster session from 9:30-10:30 and from 3:15-4:30 in the **A2P-08: MBB-P: Artificial Intelligence and Beamforming Tool** area.

**Comment by jhfournier** [comment]
Could you provide a minimum python code to create and populate a uff file  from experimental data to be used within vbeam? I only found an example writen in Matlab on Ultra Sound ToolBox.

**Reply by magnusdk** [reply]
Hi! I am working on recreating [USTB's "2D simulation of CPWC sequence with kWave"](http://www.ustb.no/html/new_example_structure/examples/kWave/CPWC_linear_array_cyst.html) using [j-Wave](https://github.com/ucl-bug/jwave) today. Let me know if you are looking for something more specific, and I will try to provide an example for that as well.

Thanks for your comment! :)

_Update: it seems I will have to finish it tomorrow 😉_

**Reply by magnusdk** [reply]
I have added [a simulation example](https://github.com/magnusdk/vbeam/blob/main/docs/examples/jwave_plane_waves_simulation.ipynb) that uses j-Wave. [You can run it on Google Colab](https://colab.research.google.com/github/magnusdk/vbeam/blob/main/docs/examples/jwave_plane_waves_simulation.ipynb), but make sure to select a GPU for the run-time.

The notebook is a bit messy _(and not exactly minimal)_ but I figured I'd share it ASAP :)

**Reply by jhfournier** [reply]
Is it possible to run "a simulation example" on a local CPU for testing? It runs well on colab, but I got the error "Cannot convert a "numpy.ndarray" to a "jaxdf.core.Field"" on my local computer on CPU. Or maybe this error is related to something else?

**Reply by magnusdk** [reply]
Hmm, I am able to run the example using a CPU 🤔 vbeam* and j-Wave depends on JAX, so you need to have that installed and `import jax.numpy as np`. Did you change any of the code in the notebook?

_*vbeam can technically run using just regular Numpy, but it is unacceptably slow for anything other than testing._

**Reply by jhfournier** [reply]
Ok i found my mistake. It runs well, thank you!

**Reply by magnusdk** [reply]
Great! Let me know if you encounter any more issues. Thank you for reaching out :)
